opentelemetry-api/src/opentelemetry/_metrics/__init__.py

```diff
@@ -132,7 +133,27 @@ def version(self):
     def schema_url(self):
         return self._schema_url
 
-    # FIXME check that the instrument name has not been used already
+    def _check_instrument_id(
```

Comment:
The API spec has [this requirement](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#:~:text=When%20more%20than%20one%20distinct%20Instrument%20is%20registered%20with%20the%20same%20name%20for%20identical%20Meters%2C%20the%20implementation%20SHOULD%20emit%20a%20warning%20to%20the%20user%20informing%20them%20of%20duplicate%20registration%20conflict(s).)
> When more than one distinct Instrument is registered with the same name for identical Meters, the implementation SHOULD emit a warning to the user informing them of duplicate registration conflict(s).

Should be calling this also in the API's no-op meter to log the warning?

Reply:
Yes, added that to the API NoOpMeter as well.